runtime/su.ts

```diff
@@ -282,32 +282,39 @@ export function mkdate(s: string): SuDate {
     return SuDate.literal(s) !;
 }
 
+enum CallType {call, callNamed, callAt}
 // Note: only Suneido values and strings are callable
-
+export let call = callFnGenerator(CallType.call);
+export let callNamed = callFnGenerator(CallType.callNamed);
+export let callAt = callFnGenerator(CallType.callAt);
 // Note: using apply until better spread (...) performance in v8
-
-export function call(f: any, ...args: any[]): any {
-    let call = f.$call;
-    if (typeof call === 'function')
-        return call.apply(f, args);
-    //  TODO strings
-    cantCall(f);
-}
-
-export function callNamed(f: any, ...args: any[]) {
-    let call = f.$callNamed;
-    if (typeof call === 'function')
-        return call.apply(f, args);
-    //  TODO strings
-    cantCall(f);
-}
-
-export function callAt(f: any, args: SuObject): any {
-    let call = f.$callAt;
-    if (typeof call === 'function')
-        return call(args);
-    //  TODO strings
-    cantCall(f);
+function callFnGenerator(type: CallType):
```

Comment:
Might be worth checking how the speed of this approach compares to writing out the three functions. I think for this code speed is more important than duplication.

Reply:
Also better to do as much work as possible outside the generated function.  e.g. '$' + CallType[type]